server/graphql/loaders/expenses.ts

```diff
@@ -1,12 +1,42 @@
 import DataLoader from 'dataloader';
 
 import ACTIVITY from '../../constants/activities';
-import models, { Op } from '../../models';
+import models, { Op, sequelize } from '../../models';
 import { ExpenseAttachedFile } from '../../models/ExpenseAttachedFile';
 import { ExpenseItem } from '../../models/ExpenseItem';
+import { LEGAL_DOCUMENT_TYPE } from '../../models/LegalDocument';
 
 import { sortResultsArray } from './helpers';
 
+const THRESHOLD = 600e2;
+const {
+  requestStatus: { RECEIVED },
+} = models.LegalDocument;
+
+
+const userTaxFormRequiredBeforePaymentQuery = `
+  SELECT 
+    e."UserId" "userId", 
+    MAX(e.id) as "expenseId", 
+    MAX(ld."requestStatus") as "legalDocRequestStatus",
+    MAX(d."documentType") as "requiredDocument",
+    SUM (amount) AS total
+  FROM "Expenses" e
+  INNER JOIN "Collectives" c ON c.id = e."CollectiveId"
+  LEFT JOIN "RequiredLegalDocuments" d ON d."HostCollectiveId" = c."HostCollectiveId"
+                                        AND d."documentType" = 'US_TAX_FORM'
+  LEFT JOIN "LegalDocuments" ld ON ld."CollectiveId" = e."FromCollectiveId"
+                                  AND ld.year = date_part('year', e."incurredAt")
+                                  AND ld."documentType" = 'US_TAX_FORM'
+  WHERE e.status IN ('PENDING', 'APPROVED', 'PAID', 'PROCESSING', 'SCHEDULED_FOR_PAYMENT')
+  AND e."UserId" IN (SELECT "UserId" FROM "Expenses" WHERE "Expenses".id IN (:expenseIds))
```

Comment:
I don't get the complexity on this line, why not directly this?

```suggestion
  AND e.id IN (:expenseIds)
```